shared/entity/type.go

```diff
@@ -57,19 +57,23 @@ var entityTypes = []Type{
 	TypeIdentityProviderGroup,
 }
 
-// String implements fmt.Stringer for Type.
-func (t Type) String() string {
-	return string(t)
+// nameToType is an internal map of TypeName to Type. It is populated on init from the types list.
+var nameToType = map[TypeName]Type{}
```

Comment:
Can we pre-allocate this using `make` either here or in `init()` to the len of types?